bundles/binding/org.openhab.binding.rwesmarthome/src/main/java/org/openhab/binding/rwesmarthome/internal/communicator/RWESmarthomeSession.java

```diff
@@ -0,0 +1,663 @@
+import org.openhab.binding.rwesmarthome.internal.communicator.util.HttpComponentsHelper;
+import org.openhab.binding.rwesmarthome.internal.communicator.util.XMLUtil;
+import org.openhab.binding.rwesmarthome.internal.communicator.xmlresponse.GetAllLogicalDeviceStatesXMLResponse;
+import org.openhab.binding.rwesmarthome.internal.communicator.xmlresponse.GetEntitiesXMLResponse;
+import org.openhab.binding.rwesmarthome.internal.communicator.xmlresponse.NotificationsXMLResponse;
+import org.openhab.binding.rwesmarthome.internal.model.Location;
+import org.openhab.binding.rwesmarthome.internal.model.LogicalDevice;
+import org.openhab.binding.rwesmarthome.internal.model.RoomTemperatureActuator;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.xml.sax.SAXException;
+
+
+/**
+ * @author ollie-dev
+ *
+ */
+public class RWESmarthomeSession {
+	
+	private static final Logger logger = LoggerFactory.getLogger(RWESmarthomeSession.class);
+	
+	/** The firmware version of RWE Smarthome */
+	public static final String FIRMWARE_VERSION = "1.70";
+	
+	/** The hostname. */
+	private String hostname = "";
+	
+	/** The username. */
+	private String username = "";
+
+	/** The password. */
+	private String password = "";
+	
+	/** The password encrypted. */
+	private String passwordEncrypted = "";
+	
+	/** The client id. */
+	private String clientId = "";
+	
+	/** The session id. */
+	private String sessionId = "";
+
+	/** The request id. */
+	private String requestId = "";
+	
+	/** The http helper. */
+	HttpComponentsHelper httpHelper = new HttpComponentsHelper();
+	
+	/** The current configuration version. */
+	private String currentConfigurationVersion = "";
+	
+	/** The locations. */
+	private ConcurrentHashMap<String, Location> locations = null;
+	
+	/** The logicalDevices */
+	private ConcurrentHashMap<String, LogicalDevice> logicalDevices = null;
+	
+	private RWEClient client;
+	
+	/**
+	 * @param client
+	 */
+	public RWESmarthomeSession(RWEClient client) {
+		super();
+		this.client = client;
+	}
+
+	/**
+	 * Returns true, if there is a valid session id.
+	 * @return boolean
+	 */
+	public boolean isValid() {
+		if (sessionId == null || "".equals(sessionId)) {
+			return false;
+		} else {
+			return true;
+		}
+	}
+
+	/**
+	 * Gets the session id.
+	 * 
+	 * @return the sessionId
+	 */
+	public String getSessionId() {
+		return sessionId;
+	}
+	
+	/**
+	 * @return the locations
+	 */
+	public ConcurrentHashMap<String, Location> getLocations() {
+		return locations;
+	}
+
+	/**
+	 * @return the logicalDevices
+	 */
+	public ConcurrentHashMap<String, LogicalDevice> getLogicalDevices() {
+		return logicalDevices;
+	}
+
+	/**
+	 * Destroy.
+	 */
+	public void destroy() {
+		String logoutrequest = "";
+		logoutrequest = "<BaseRequest xmlns:xsi=\"http://www.w3.org/2001/XMLSchema-instance\" xsi:type=\"LogoutRequest\" Version=\"" + FIRMWARE_VERSION + "\" RequestId=\""
+				+ requestId + "\" SessionId=\"" + getSessionId() + "\" />";
```

Comment:
better collect all XML-Snippets as constants with substitution marks and use String.format() in order to replace these marks. This will increase code readability (also change for all subsequent XML Snippets)